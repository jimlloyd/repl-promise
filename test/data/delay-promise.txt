function delay(ms) {
    var setTimeout = require('timers').setTimeout;
    var deferred = Q.defer();
    function foo() {
        deferred.resolve({
            wokeafter: ms,
        });
    }
    setTimeout(foo, ms);
    return deferred.promise;
}
delay(10)
delay(20)
delay(5)
null
undefined
"Goodbye"
