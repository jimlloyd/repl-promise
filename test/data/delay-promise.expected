node > function delay(ms) {
...     var setTimeout = require('timers').setTimeout;
...     var deferred = Q.defer();
...     function foo() {
...         deferred.resolve({
...             wokeafter: ms,
...         });
...     }
...     setTimeout(foo, ms);
...     return deferred.promise;
... }
node > delay(10)
{ wokeafter: 10 }
node > delay(20)
{ wokeafter: 20 }
node > delay(5)
{ wokeafter: 5 }
node > null
node > undefined
node > "Goodbye"
'Goodbye'
